### ISSUE TITLE
On particular map the AI seems to not pick up resources

### ISSUE THREAD

**Issue by crypt47** [ihhub/fheroes2#8586]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Other

### Describe the bug

On some maps or in some cases AI leaves resources on the ground. Please, check out the save file. The hero Astra finds a captured mine surrounded by resouces directly on the aquire path. Doesn't make much sense.

### Save file

[Seven_Lakes_0022.sav.zip](https://github.com/ihhub/fheroes2/files/15092393/Seven_Lakes_0022.sav.zip)

**Comment by oleg-derevenetz** [comment]
The AI may well not collect resources if it believes that the value of the resource for its castle is relatively low, and it believes that it has some task with a higher priority. Now it's hard to say exactly what distracted him from collecting this crystal pile because AI hero has already left there.

**Comment by crypt47** [comment]
> The AI may well not collect resources if it believes that the value of the resource for its castle is relatively low, and it believes that it has some task with a higher priority. Now it's hard to say exactly what distracted him from collecting this crystal pile because AI hero has already left there.

the thing is it's pretty common all over this map. Sometimes it picks, sometimes not. Well, strategically it doesn't make sense, because somebody else will pick.

**Comment by oleg-derevenetz** [comment]
> Any other ways for me to study this without builiding the debug version?

No, you will at least need the debug build with verbose AI logging enabled (to see how AI selects targets for heroes - with numbers). And also maybe a debugger too.

**Comment by crypt47** [comment]
> > Any other ways for me to study this without builiding the debug version?
> 
> No, you will at least need the debug build with verbose AI logging enabled (to see how AI selects targets for heroes - with numbers). And also maybe a debugger too.

Ok, thank you for your answer. I'm not going to study it right now, but it's good to know what to do.

**Comment by zenseii** [comment]
This issue lacks the necessary information to check it. The AI isn't supposed to mindlessly waste it's time to pick up every resource pile.

Sure, there are times when negating your enemy of resources when you have enough can be a viable strategy, but saving movement points is often a much better choice for the AI, especially considering that it doesn't consider creature speeds when including creatures in its army.

Personally I think we can close this issue and reopen it if a way to reproduce it is provided.

**Comment by ihhub** [comment]
I am closing this issue as I don't see a particular bug in the AI behavior.
